hypothesis-python/src/hypothesis/internal/conjecture/junkdrawer.py

```diff
@@ -116,11 +116,13 @@ def __len__(self) -> int:
     def __getitem__(self, i: int) -> int: ...  # pragma: no cover
 
     @overload
-    def __getitem__(self, i: slice) -> "IntList": ...  # pragma: no cover
+    def __getitem__(
+        self, i: slice
+    ) -> Union[list[int], ArrayType[int]]: ...  # pragma: no cover
 
-    def __getitem__(self, i: Union[int, slice]) -> "Union[int, IntList]":
-        if isinstance(i, slice):
-            return IntList(self.__underlying[i])
+    def __getitem__(
+        self, i: Union[int, slice]
+    ) -> Union[int, list[int], ArrayType[int]]:
         return self.__underlying[i]
```

Comment:
We don't actually getitem with a slice anywhere except tests. If we ever do, hopefully mypy will catch us accidentally relying on an IntList return. I think this is a fine tradeoff for performance.